### ISSUE TITLE
[Bug] Stardist fail always at around 95%

### ISSUE THREAD

**Issue by PietroD** [gustaveroussy/sopa#367]
I am trying to run Stardsit on CosMx data via snakemake but keeps failing while almost done on segmentation step.

Here is the log:

[2025-12-04T105728.497313.snakemake.log](https://github.com/user-attachments/files/23955043/2025-12-04T105728.497313.snakemake.log)

And here is the log for the last patch failing:

[2025-12-04T105728.497313.snakemake.log](https://github.com/user-attachments/files/23955049/2025-12-04T105728.497313.snakemake.log)

Help would be really appreciated as always.
Thanks a lot 

## System
 - Python version 3.10.16
<details>
<summary>Dependencies versions</summary>
<br>
Package                       Version
----------------------------- --------------
absl-py                       2.1.0
aiobotocore                   2.19.0
aiohappyeyeballs              2.4.6
aiohttp                       3.11.12
aioitertools                  0.12.0
aiosignal                     1.3.2
alabaster                     1.0.0
alphashape                    1.3.1
anndata                       0.11.3
annotated-types               0.7.0
anyio                         4.9.0
app-model                     0.3.1
appdirs                       1.4.4
argon2-cffi                   23.1.0
argon2-cffi-bindings          21.2.0
array_api_compat              1.10.0
arrow                         1.3.0
asciitree                     0.3.3
asttokens                     3.0.0
astunparse                    1.6.3
async-lru                     2.0.5
async-timeout                 5.0.1
attrs                         25.1.0
babel                         2.17.0
beautifulsoup4                4.13.3
bleach                        6.2.0
botocore                      1.36.3
build                         1.2.2.post1
cachey                        0.2.1
cellpose                      3.1.1.1
certifi                       2025.1.31
cffi                          1.17.1
charset-normalizer            3.4.1
click                         8.1.8
click-log                     0.4.0
cloudpickle                   3.1.1
colorcet                      3.1.0
comm                          0.2.2
ComSeg                        1.8.5
contourpy                     1.3.1
csbdeep                       0.8.1
cycler                        0.12.1
dask                          2024.11.2
dask-expr                     1.1.19
dask-image                    2024.5.3
datashader                    0.17.0
debugpy                       1.8.14
decorator                     5.2.1
defusedxml                    0.7.1
distributed                   2024.11.2
docrep                        0.3.2
docstring_parser              0.16
docutils                      0.21.2
et_xmlfile                    2.0.0
exceptiongroup                1.2.2
executing                     2.2.0
fasteners                     0.19
fastjsonschema                2.21.1
fastremap                     1.15.1
filelock                      3.18.0
flake8                        7.1.2
flatbuffers                   25.2.10
flexcache                     0.3
flexparser                    0.4
FlowIO                        1.3.0
fonttools                     4.56.0
fqdn                          1.5.1
freetype-py                   2.5.1
frozenlist                    1.5.0
fsspec                        2025.2.0
gast                          0.6.0
geopandas                     1.0.1
google-pasta                  0.2.0
grpcio                        1.70.0
h11                           0.14.0
h5py                          3.12.1
harmonypy                     0.0.10
HeapDict                      1.0.1
hsluv                         5.0.4
httpcore                      1.0.7
httpx                         0.28.1
idna                          3.10
igraph                        0.11.8
imagecodecs                   2024.12.30
imageio                       2.37.0
imagesize                     1.4.1
importlib_metadata            8.6.1
in-n-out                      0.2.1
inflect                       7.5.0
ipykernel                     6.29.5
ipython                       8.37.0
isoduration                   20.11.0
jedi                          0.19.2
Jinja2                        3.1.5
jmespath                      1.0.1
joblib                        1.4.2
json5                         0.10.0
jsonpointer                   3.0.0
jsonschema                    4.23.0
jsonschema-specifications     2024.10.1
jupyter_client                8.6.3
jupyter_core                  5.8.1
jupyter-events                0.12.0
jupyter-lsp                   2.2.5
jupyter_server                2.15.0
jupyter_server_terminals      0.5.3
jupyterlab                    4.3.6
jupyterlab_pygments           0.3.0
jupyterlab_server             2.27.3
keras                         3.8.0
kiwisolver                    1.4.8
lazy_loader                   0.4
legacy-api-wrap               1.4.1
leidenalg                     0.10.2
libclang                      18.1.1
llvmlite                      0.44.0
locket                        1.0.0
loguru                        0.7.3
loompy                        3.0.8
magicgui                      0.10.0
Markdown                      3.7
markdown-it-py                3.0.0
MarkupSafe                    3.0.2
matplotlib                    3.10.0
matplotlib-inline             0.1.7
matplotlib-scalebar           0.9.0
mccabe                        0.7.0
mdurl                         0.1.2
mistune                       3.1.3
ml-dtypes                     0.4.1
more-itertools                10.7.0
mpmath                        1.3.0
msgpack                       1.1.0
multidict                     6.1.0
multipledispatch              1.0.0
multiscale_spatial_image      2.0.2
namex                         0.0.8
napari                        0.5.6
napari-console                0.1.3
napari-matplotlib             3.0.0
napari-plugin-engine          0.2.0
napari-spatialdata            0.5.5
napari-svg                    0.2.1
natsort                       8.4.0
nbclient                      0.10.2
nbconvert                     7.16.6
nbformat                      5.10.4
nest-asyncio                  1.6.0
networkx                      3.4.2
notebook_shim                 0.2.4
npe2                          0.7.8
numba                         0.61.0
numcodecs                     0.13.1
numpy                         1.26.4
numpy-groupies                0.11.2
numpydoc                      1.8.0
nvidia-cublas-cu12            12.4.5.8
nvidia-cuda-cupti-cu12        12.4.127
nvidia-cuda-nvrtc-cu12        12.4.127
nvidia-cuda-runtime-cu12      12.4.127
nvidia-cudnn-cu12             9.1.0.70
nvidia-cufft-cu12             11.2.1.3
nvidia-curand-cu12            10.3.5.147
nvidia-cusolver-cu12          11.6.1.9
nvidia-cusparse-cu12          12.3.1.170
nvidia-cusparselt-cu12        0.6.2
nvidia-nccl-cu12              2.21.5
nvidia-nvjitlink-cu12         12.4.127
nvidia-nvtx-cu12              12.4.127
ome-types                     0.5.3
ome-zarr                      0.10.3
omnipath                      1.0.12
opencv-python                 4.11.0.86
opencv-python-headless        4.11.0.86
openpyxl                      3.1.5
opt_einsum                    3.4.0
optree                        0.14.0
overrides                     7.7.0
packaging                     24.2
pandas                        2.2.3
pandocfilters                 1.5.1
param                         2.2.0
parso                         0.8.4
partd                         1.4.2
patsy                         1.0.1
pexpect                       4.9.0
pickleshare                   0.7.5
pillow                        11.1.0
PIMS                          0.7
Pint                          0.24.4
pip                           25.0.1
platformdirs                  4.3.6
pooch                         1.8.2
prometheus_client             0.21.1
prompt_toolkit                3.0.51
propcache                     0.2.1
protobuf                      5.29.3
psutil                        7.0.0
psygnal                       0.12.0
ptyprocess                    0.7.0
pure_eval                     0.2.3
pyarrow                       19.0.0
pycodestyle                   2.12.1
pyconify                      0.2.1
pycparser                     2.22
pyct                          0.5.0
pydantic                      2.10.6
pydantic-compat               0.1.2
pydantic_core                 2.27.2
pyflakes                      3.2.0
Pygments                      2.19.1
pynndescent                   0.5.13
pyogrio                       0.10.0
PyOpenGL                      3.1.9
pyparsing                     3.2.1
pyproj                        3.7.1
pyproject_hooks               1.2.0
PyQt5                         5.15.11
PyQt5-Qt5                     5.15.16
PyQt5_sip                     12.17.0
pyqtgraph                     0.13.7
python-dateutil               2.9.0.post0
python-json-logger            3.3.0
pytz                          2025.1
PyYAML                        6.0.2
pyzmq                         27.0.0
qtconsole                     5.6.1
QtPy                          2.4.3
readfcs                       2.0.1
referencing                   0.36.2
requests                      2.32.3
rfc3339-validator             0.1.4
rfc3986-validator             0.1.1
rich                          13.9.4
roifile                       2025.2.20
rpds-py                       0.24.0
rtree                         1.4.0
s3fs                          2025.2.0
scanpy                        1.11.0
scikit-image                  0.25.1
scikit-learn                  1.5.2
scikit-misc                   0.5.1
scipy                         1.15.2
seaborn                       0.13.2
Send2Trash                    1.8.3
session-info2                 0.1.2
setuptools                    75.8.0
shapely                       2.0.7
shellingham                   1.5.4
six                           1.17.0
slicerator                    1.1.0
sniffio                       1.3.1
snowballstemmer               2.2.0
sopa                          2.0.4
sortedcontainers              2.4.0
soupsieve                     2.6
spatial_image                 1.1.0
spatialdata                   0.3.0
spatialdata-io                0.1.7
spatialdata-plot              0.2.9
Sphinx                        8.1.3
sphinxcontrib-applehelp       2.0.0
sphinxcontrib-devhelp         2.0.0
sphinxcontrib-htmlhelp        2.1.0
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          2.0.0
sphinxcontrib-serializinghtml 2.0.0
squidpy                       1.6.5
stack-data                    0.6.3
stardist                      0.9.1
statannot                     0.2.3
statsmodels                   0.14.4
superqt                       0.7.2
sympy                         1.13.1
tabulate                      0.9.0
tangram-sc                    1.0.4
tblib                         3.0.0
tensorboard                   2.18.0
tensorboard-data-server       0.7.2
tensorflow                    2.18.0
tensorflow-io-gcs-filesystem  0.37.1
termcolor                     2.5.0
terminado                     0.18.1
texttable                     1.7.0
threadpoolctl                 3.5.0
tifffile                      2025.1.10
tinycss2                      1.4.0
toml                          0.10.2
tomli                         2.2.1
tomli_w                       1.2.0
toolz                         1.0.0
torch                         2.6.0
tornado                       6.5.1
tqdm                          4.67.1
traitlets                     5.14.3
trimesh                       4.6.12
triton                        3.2.0
typeguard                     4.4.4
typer                         0.15.1
types-python-dateutil         2.9.0.20241206
typing_extensions             4.14.0
tzdata                        2025.1
umap-learn                    0.5.7
uri-template                  1.3.0
urllib3                       2.3.0
validators                    0.35.0
vispy                         0.14.3
wcwidth                       0.2.13
webcolors                     24.11.1
webencodings                  0.5.1
websocket-client              1.8.0
Werkzeug                      3.1.3
wheel                         0.45.1
wrapt                         1.17.2
xarray                        2024.11.0
xarray-dataclasses            1.9.1
xarray-datatree               0.0.14
xarray-schema                 0.0.3
xarray-spatial                0.4.0
xsdata                        24.3.1
yarl                          1.18.3
zarr                          2.18.3
zict                          3.0.0
zipp                          3.21.0
</details>
